### PR TITLE
KE2: Extract `bool`, `char`, `float`, `double` constants

### PR DESCRIPTION
**Pull request by tamasvajk**

### CODE REVIEW

**Review by igfoo — APPROVED** [review]
Otherwise LGTM

File: java/kotlin-extractor2/src/main/kotlin/entities/Expression.kt [hunk 1]
```diff
@@ -217,6 +217,47 @@ private fun KotlinFileExtractor.extractNull(
...
+    OLD: KE1
+            overrideId: Label<out DbExpr>? = null
+    */
+) =
+    // OLD: KE1: Was: exprIdOrFresh<DbIntegerliteral>(overrideId).also {
+    tw.getFreshIdLabel<T>().also {
+        val type = useType(t)
+        write(it, type.javaResult.id, parent, idx)
+        tw.writeExprsKotlinType(it, type.kotlinResult.id)
+        tw.writeNamestrings(textName, textValue, it as Label<out DbNamedexprorstmt>)
```
[the remark this answers is not in the document]
Comment (tamasvajk) [reply java/kotlin-extractor2/src/main/kotlin/entities/Expression.kt:257]: That doesn't work with `tw.writeExprsKotlinType` and `extractExprContext`.

[the remark this answers is not in the document]
Comment (igfoo) [reply java/kotlin-extractor2/src/main/kotlin/entities/Expression.kt:270]: Ah yes, sorry, it should be `private fun <T : DbLiteral>`.

[the remark this answers is not in the document]
Comment (tamasvajk) [reply java/kotlin-extractor2/src/main/kotlin/entities/Expression.kt]: Thanks.

**Review by igfoo — APPROVED** [review]

**Pull request merged**